src/Sylius/Component/User/Model/User.php

```diff
@@ -0,0 +1,670 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * This component was inspired by FOS User-Bundle
+ */
+
+namespace Sylius\Component\User\Model;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+
+/**
+ * @author Paweł Jędrzejewski <pawel@sylius.org>
+ * @author Łukasz Chruściel <lukasz.chrusciel@lakion.com>
+ * @author Michał Marcinkowski <michal.marcinkowski@lakion.com>
+ */
+class User implements UserInterface
+{
+    /**
+     * @var int
+     */
+    protected $id;
+
+    /**
+     * @var CustomerInterface
+     */
+    protected $customer;
+
+    /**
+     * @var string
+     */
+    protected $username;
```

Comment:
Right now I've created listener (DefaultUsernameListener) which synchronizes the user's username with it's customer's email. Default user provider queries by username or email (you can change it), so it is totally possible to turn off this listener and use username as login indentifier or anything you want. It is not enforcing anything and is flexible as much as whole Sylius is :)